### PR TITLE
engine: support static rebuilds

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism, @maiamcc,

Please review the following commits I made in branch nicks/ch390/watch:

4c9968773b98a6ad7d093147c35cd0b2c613338b (2018-09-27 12:00:13 -0400)
engine: support static rebuilds

### CODE REVIEW

File: internal/engine/upper_test.go [hunk 1]
```diff
@@ -391,6 +391,36 @@ func TestRebuildWithSpuriousChangedFiles(t *testing.T) {
...
+	go func() {
+		call := <-f.b.calls
+		assert.True(t, call.state.IsEmpty())
+
+		// Simulate a change to main.go
+		mainPath := filepath.Join(f.Path(), "main.go")
+		f.watcher.events <- watch.FileEvent{Path: mainPath}
+
+		// The next build should treat main.go as changed, and build
+		// on the last successful result, from before a.go changed.
```
Comment (maiamcc) [inline internal/engine/upper_test.go]: this comment isn't relevant to this test

Comment (maiamcc) [reply internal/engine/upper_test.go]: ...also am i wrong, or are changed files not really relevant to a static build (except that the watcher tells us that something has changed)?

Comment (nicks) [reply internal/engine/upper_test.go]: updated!

no, static builds still trigger a rebuild when the file changes. i don't think the name "static" build is a good one.

Comment (maiamcc) [reply internal/engine/upper_test.go]: ah sorry, i mean: it's true that a file change triggers a rebuild, but that that information (i.e. the file(s) changed) isn't used anywhere in the build itself -- i.e. `call.state.FilesChanged()` is irrelevant to the building. right?

**Pull request merged**